### PR TITLE
[App/Improvement] Cleaning up Queue abstraction

### PR DESCRIPTION
**Pull request by hhsecond**
## What does this PR do?

We needed to clean up the queue abstraction to introduce the HTTP queue. This small PR needed to go first for simultaneously making further changes to both cloud and framework

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning_app/core/queues.py [hunk 1]
```diff
@@ -149,6 +149,14 @@ def get(self, timeout: int = None):
         """
         pass
 
+    @property
+    def is_running(self) -> bool:
+        """Returns True if the queue is running, False otherwise.
+
+        Child classes should override this property and implement custom logic as requires
```
Comment (awaelchli) [inline src/lightning_app/core/queues.py:156]:
```suggestion
        Child classes should override this property and implement custom logic as required
```

File: src/lightning_app/core/constants.py [hunk 2]
```diff
@@ -5,6 +5,26 @@
 
 import lightning_app
 
+
+def truthy(val) -> bool:
```
Comment (awaelchli) [inline src/lightning_app/core/constants.py:9]: Where is this used in this PR?

**Review by awaelchli — APPROVED** [review]

File: src/lightning_app/core/constants.py (hunk 2 above)
Comment (hhsecond) [reply src/lightning_app/core/constants.py:9]: this was a leak from my PR division. I'll add it to the right PR. Thank you

**Review by manskx — APPROVED** [review]

**Pull request merged**